Lib/fractions.py

```diff
@@ -567,11 +567,14 @@ def __hash__(self):
         # dinv is the inverse of self._denominator modulo the prime
         # _PyHASH_MODULUS, or 0 if self._denominator is divisible by
         # _PyHASH_MODULUS.
-        dinv = pow(self._denominator, _PyHASH_MODULUS - 2, _PyHASH_MODULUS)
+        try:
+            dinv =  pow(self._denominator, -1, _PyHASH_MODULUS)
+        except ValueError:
+            dinv = pow(self._denominator, _PyHASH_MODULUS - 2, _PyHASH_MODULUS)
```

Comment:
`ValueError` is raised if and only if `_PyHASH_MODULUS` divides `self._denominator`, in which case the second `pow()` will just return 0.  So this line could be replaced with `dinv = 0`, preferably with a comment to the effect that there is no modular inverse in this branch.

Reply:
Done.